ui.go

```diff
@@ -441,49 +441,43 @@ func (win *win) printDir(screen tcell.Screen, dir *dir, context *dirContext, dir
 			}
 		}
 
+		// make space for select marker, and leave another space at the end
+		maxWidth := win.w - lnwidth - 2
+		// make extra space to separate windows if drawbox is not enabled
+		if !gOpts.drawbox {
```

Comment:
(No action needed) I don't think we need this, printing one more letter when drawbox is off seems like it would be better. One of us can try it out and change it in a future PR.

Reply:
Even if `drawbox` isn't enabled, I feel you would still want to keep the panes physically separate (e.g. separate the cursor in one pane from the line number in the next pane), which is what the space is for.

I'm quite sure this was the intended logic even before I started working on this code. TBH I'm sort of hesitant to change any logic relating to the layout or formatting of UI components, in case someone complains about it later on. You are of course welcome to experiment and see if you can come up with something better though.

Reply:
>  I feel you would still want to keep the panes physically separate (e.g. separate the cursor in one pane from the line number in the next pane),

Yes, but it seemed to me one space (the "another space in the end" in the comment above this line) should be enough. Or is that character used for some purpose I missed?

> I'm quite sure this was the intended logic even before I started working on this code.

Oh, I hadn't realized you added this in https://github.com/gokcehan/lf/pull/1150. I hadn't reviewed that PR carefully and haven't checked myself what was happening before then, but I believe you.

Currently, `:set drawbox!` does not affect the maximum width allowed for filenames. To me, it seems reasonable to allow one more character when the lines don't take up a character.

Either way, I don't think this is a huge problem.

> You are of course welcome to experiment and see if you can come up with something better though.

Yeah, I'll make some screenshots when I have a moment.

Reply:
Also, while strictly speaking what I'm proposing is a change to the UI, it is meant to be extremely mild: one more character printed for long file names. If all filenames are extremely long, the result would be that one space separates the columns as opposed to two spaces.

Reply:
It seems like I was wrong here as well (as in my other comment). Two spaces are important when the cursor is on the file. One of them pads the filename inside the cursor, while the other one separates the columns.

Reply:
> One of them pads the filename inside the cursor

The "another space in the end" part was existing logic, and it doesn't serve any special purpose like selection or tag markers, and I also came to the same conclusion that it's just for padding. I do like it though - most of the time files aren't tagged, so it makes the cursor look balanced.

Looks like only the PR author can mark a conversation as resolved, I'll do that then.

Reply:
Yeah, the cursor looks much better with the padding spaces on both sides.